### PR TITLE
Fix CreatedAt state

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

File: ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs [hunk 1]
```diff
@@ -23,6 +23,18 @@ public override async Task<List<AssistanceCategoryEntity>> GetAllAsync(Cancellat
...
                 .ToListAsync(cancellationToken);
         }
+        public async override Task<AssistanceCategoryEntity> UpdateAsync(AssistanceCategoryEntity entity, CancellationToken cancellationToken)
+        {
+            var existingEntity = await Query.AsNoTracking()
+                .FirstOrDefaultAsync(e => e.Id == entity.Id, cancellationToken);
+
+
+            entity.CreatedAt = existingEntity?.CreatedAt ?? entity.CreatedAt;
+
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs:33]: Don't get why you need this line if you have interceptor

File: ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceRepository.cs [hunk 2]
```diff
@@ -45,6 +45,7 @@ public async override Task<AssistanceEntity> UpdateAsync(AssistanceEntity entity
 
             entity.UserProfileId = existingEntity?.UserProfileId ?? entity.UserProfileId;
             entity.AssistanceCategoryId = existingEntity?.AssistanceCategoryId ?? entity.AssistanceCategoryId;
+            entity.CreatedAt = existingEntity?.CreatedAt ?? entity.CreatedAt;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceRepository.cs:48]: same

File: ServiceFinder/ServiceFinder.DAL/Repositories/UserProfileRepository.cs [hunk 3]
```diff
@@ -26,5 +26,16 @@ public override async Task<List<UserProfileEntity>> GetAllAsync(CancellationToke
                 .Include(u => u.Reviews)
                 .ToListAsync(cancellationToken);
         }
+        public async override Task<UserProfileEntity> UpdateAsync(UserProfileEntity entity, CancellationToken cancellationToken)
+        {
+            var existingEntity = await Query.AsNoTracking()
+                .FirstOrDefaultAsync(e => e.Id == entity.Id, cancellationToken);
+
+            entity.CreatedAt = existingEntity?.CreatedAt ?? entity.CreatedAt;
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Repositories/UserProfileRepository.cs:34]: same

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs (hunk 1 above)
Comment (vladzhorov) [reply ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs:33]: forgot(

File: ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs [hunk 4]
```diff
@@ -24,5 +24,15 @@ public override async Task<PagedResult<AssistanceCategoryEntity>> GetAllAsync(in
             var query = Query.Include(ac => ac.Assistances);
             return await GetPagedResultAsync(query, pageNumber, pageSize, cancellationToken);
         }
+
+        public async override Task<AssistanceCategoryEntity> UpdateAsync(AssistanceCategoryEntity entity, CancellationToken cancellationToken)
+        {
+            var existingEntity = await Query.AsNoTracking()
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Repositories/AssistanceCategoryRepository.cs:30]: Remove the unused local variable 'existingEntity'. (https://rules.sonarsource.com/csharp/RSPEC-1481)

File: ServiceFinder/ServiceFinder.DAL/Repositories/UserProfileRepository.cs [hunk 5]
```diff
@@ -26,5 +26,15 @@ public override async Task<PagedResult<UserProfileEntity>> GetAllAsync(int pageN
             var query = Query.Include(u => u.Assistances).ThenInclude(a => a.Reviews).Include(u => u.Reviews);
             return await GetPagedResultAsync(query, pageNumber, pageSize, cancellationToken);
         }
+        public async override Task<UserProfileEntity> UpdateAsync(UserProfileEntity entity, CancellationToken cancellationToken)
+        {
+            var existingEntity = await Query.AsNoTracking()
```
Comment (moltenessence) [inline ServiceFinder/ServiceFinder.DAL/Repositories/UserProfileRepository.cs:31]: Remove the unused local variable 'existingEntity'.

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by pavelGagarin0870 — APPROVED** [review]

**Review by moltenessence — APPROVED** [review]

**Pull request merged**